manifests/install.pp

```diff
@@ -2,10 +2,25 @@
 class nats::install {
   contain "nats::install::${nats::service_type}"
 
+  $_manage_group = $nats::manage_group and $nats::group != "root"
+  if $_manage_group {
```

Comment:
PR looks good - apart from this, kill this temp var as well and just do `if $nats::manage_group and $nats::group != "root" {` and add a new line between this if and the one below.

I'll need a issue opened for this, once you have the number please squash this into one command and change the commit format to `(#n) Add option to run NATS as a user`